app/frontend/components/pages/staking/stakingPage.js

```diff
@@ -0,0 +1,103 @@
+const {h, Component} = require('preact')
+const connect = require('unistore/preact').connect
+const actions = require('../../../actions')
+
+class StakingPage extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      email: '',
+      emailValid: false,
+      errorMessage: '',
+    }
+
+    this.isValidEmail = this.isValidEmail.bind(this)
+    this.updateEmail = this.updateEmail.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+  }
+
+  isValidEmail(email) {
+    // eslint-disable-next-line max-len
+    const re = /^[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?$/gim
+    return re.test(email)
+  }
+
+  updateEmail(e) {
+    const isEmailValid = this.isValidEmail(e.target.value)
+    this.setState({
+      email: e.target.value,
+      emailValid: isEmailValid,
+      errorMessage: !isEmailValid && 'Invalid email format',
+    })
+    this.props.resetEmailSubmission()
+  }
+
+  handleSubmit(e) {
+    e.preventDefault()
+    if (this.state.emailValid) this.props.submitEmail(this.state.email)
+  }
+
+  render({emailSubmitSuccess, emailSubmitMessage}, {email, emailValid, errorMessage}) {
+    return h(
+      'div',
+      {class: 'staking-wrapper'},
+      h(
+        'div',
+        {class: 'staking-inner'},
+        h('div', {class: 'staking-label'}, 'Upcoming'),
+        h(
+          'h2',
+          {class: 'staking-title'},
+          'Staking delegation and staking pool is coming to AdaLite'
+        ),
+        h(
+          'p',
+          {class: 'staking-text'},
+          'We are planing to implement staking delegation interface so our users can easily stake their ADA to any stakepool directly from AdaLite. We also plan to operate our own AdaLite stake pool with reasonable fees and we hope AdaLite users will be willing to stake with us.'
+        ),
+        h(
+          'form',
+          {
+            class: 'staking-form',
+            id: 'stakingForm',
+          },
+          h('input', {
+            class: 'input',
+            type: 'email',
+            placeholder: 'Enter your email to get notified',
+            value: email,
+            required: true,
+            onInput: this.updateEmail,
```

Comment:
Currently the error message shows as soon as user starts typing the email, saying it has invalid format even though the user is clearly not done typing and the error message stays displayed even though the user clears the input. IMO update email `onBlur` and remove the `disabled` property from submitButton altogether. This way, no error message is shown until a user tries to submit. What do you think?